vowpalwabbit/parse_regressor.cc

```diff
@@ -327,19 +327,21 @@ void save_load_header(
             bytes_read_write += bin_text_read_write_fixed_validated(
                 model_file, (char*)&inter_len, sizeof(inter_len), "", read, msg, text);
             if (!read)
-            {
-              memcpy(buff2, all.interactions[i].c_str(), inter_len);
+            { 
+              std::string s(all.interactions[i].begin(),all.interactions[i].end());
+              memcpy(buff2, s.c_str(), inter_len);
 
               msg << "interaction: ";
-              msg.write(all.interactions[i].c_str(), inter_len);
+              msg.write(s.c_str(), inter_len);
             }
 
             bytes_read_write += bin_text_read_write_fixed_validated(model_file, buff2, inter_len, "", read, msg, text);
 
             if (read)
             {
               std::string temp(buff2, inter_len);
-              all.interactions.push_back(temp);
+              std::vector<uint8_t> temp2(temp.begin(), temp.end());
```

Comment:
Can you remove this temporary?

Reply:
I've removed temp2 and I've used emplace_back here instead now

Reply:
Removed